doc/extension_store/README.md

```diff
@@ -0,0 +1,76 @@
+    * `BUCKET`
+* Variables prompted to users are defined in the `parameters` section. The Extension Store asks the user to input values for all parameters.
+* Variable placeholders in strings (e.g. BucketFS paths or SQL scripts) are expanded before using them.
+
+## Parameters to Store
+
+The Extension management component on the cluster must store the following information for an installed VS:
+
+* For each configuration:
+    * ID and version
+        * Required to detect if an older version is already installed
+        * Required to detect if the user tries to install another VS using the same configuration
+    * Paths to installed libraries and configuration files on BucketFS
+        * File names might change in newer versions.
+        * Needed to delete files from BucketFS during update/delete
+    * Parameters with `scope=configuration`
+        * Required when creating a new VS using the same configuration.
+        * Cannot be changed during update. Maybe hide them?
+* For each Connection / Virtual Schema:
+    * ID and version of the configuration
+        * Required to check if another VS uses the same configuration
+    * Parameters with `scope=connection` that the user entered during the installation process, required for updating the values.
+    * **Don't store** values for fields with `type=password`.
+        * User has to enter a new password when updating.
+
+## Installation Process
+
+1. If no other VS uses this configuration or the configuration was not installed before:
+    1. Ask user to enter values for all parameters with `scope=configuration`
+    1. If `licenseAgreementRequired=true` (usually for JDBC drivers): show the license and ask user to agree
+    1. Download all artifacts in the `bucketFsUpload`
+    1. Upload the artifacts to BucketFS, using the specified path names
+    1. Run the `create.configuration` SQL statements
+1. Ask user to enter values for all parameters with `scope=connection`
+1. Run the `create.connection` SQL statements.
+
+## Update Process
+
+1. Check if there is a newer version of the configuration. If yes:
+    1. Ask user to confirm the license again (usually the JDBC driver license)
+    1. Download new JARs
+    1. Delete old JARs from BucketFS
+    1. Install new JARs to BucketFS
+    1. Run scripts from `update.configuration`
+1. Let user enter new values for all parameters with `scope=connection`
+1. Run `update.connection` SQL statements.
+
+## Delete Process
+
+1. Run `drop.connection` SQL statements.
+1. If no other VS uses this configuration:
+    1. Run `drop.configuration` SQL statements.
+    1. Delete JARs from BucketFS (using stored paths)
+
+## Open Points
+
+* Current JSON structure and property names is just a draft and open for discussion!
+* What happens if the Adapter Schema or other resources already exists?
+* Simplify process by auto-generating names, e.g. `connectionName`, `adapterScriptName`.
+* How to detect if another VS already has installed some JARs?
```

Comment:
If each extension gets its own directory and only uses this, you probably don't need to care about duplicated jars. I see deduplcation as an optimization.